### DISCUSSION TITLE
Run standalone/docker-compose.yml with Podman

### DISCUSSION THREAD

**Discussion by chopeen** [milvus-io/milvus#30228] [Show and tell]
I followed the article [Using Compose Files with Podman](https://docs.oracle.com/en/learn/podman-compose/index.html#confirm-docker-compose-is-working) and installed `podman-compose` on a RHEL machine.

`podman-compose up -d` was working fine with [docker-compose.yml](https://github.com/milvus-io/milvus/blob/c9860e3d2a904124d82d6909cf9df0a33599dc70/deployments/docker/standalone/docker-compose.yml), but the Milvus `standalone` container was not starting because of error like:

```
last connection error: connection error: 
desc = \"transport: Error while dialing: dial tcp: lookup etcd on 145.245.102.102:53: no such host
```

Basically, it could not connect to the `etcd` host and it was trying to use regular DNS to resolve this name.

--

Running `podman network inspect milvus` showed the following:

```
[
     {
          "name": "milvus",
          ...
          "ipv6_enabled": false,
          "internal": false,
          "dns_enabled": false,
          ...
          "ipam_options": {
               "driver": "host-local"
          }
     }
]
```

`"dns_enabled": false` is the problem's root cause.

---

Installing `podman-plugins` to have the `dnsname` plugin fixed the issue.

Source: [configuring-podman-for-inter-container-communication](https://www.insentragroup.com/us/insights/geek-speak/modern-workplace/configuring-podman-for-inter-container-communication)

**Comment by yhmo** [comment]
Useful experience.

**Comment by pai4451** [comment]
Very helpful
